cli/main.rs

```diff
@@ -32,9 +35,19 @@ async fn main_impl() -> Result<()> {
                 &file,
                 &std::env::current_dir().context("unable to get current working directory")?,
             )?;
+
+            let lassie_daemon = Arc::new(
+                lassie::Daemon::start(lassie::DaemonConfig {
+                    temp_dir: None, // TODO: Should we use something like ~/.cache/zinnia/lassie?
```

Comment:
Station Core will pass `$CACHE_ROOT`:

https://github.com/filecoin-station/core/blob/7f07e5203c71366fa9bde713b0d28dee9ea0c51d/lib/zinnia.js#L48-L55.

Can we make Zinnia use that if it is set?

Reply:
I am already using CACHE_ROOT in `zinniad`, see here:

https://github.com/filecoin-station/zinnia/pull/243/files#diff-c74dac62db9c80f1be22978c93249f7b304e05ddb38131e7969efa13effaeb1eR45

The file `cli/main.js` implements `zinnia`, the CLI people use locally when building Station modules. Let's explore together what a good developer experience would look like?

IMO:
- We should not force `zinnia` users to always provide `CACHE_ROOT`. We don't ask them for `FIL_WALLET_ADDRESS` either. This way, users can type `zinnia run main.js`, and all works out of the box.
- I guess allowing CLI users to control the CACHE ROOT can be helpful. I am not sure, though, if an env var provides good ergonomy. Would a project-specific config file be easier to use?
- How important is this? Can we leave the current solution and open a follow-up GH issue to discuss what would a good (and easy-to-implement) solution look like?
- Note: if we tell Lassie to use a specific temp dir that's not automatically cleaned by the operating system, we will need to clean any leftover files ourselves.

Reply:
> Note: if we tell Lassie to use a specific temp dir that's not automatically cleaned by the operating system, we will need to clean any leftover files ourselves.

I'll be implementing this cleanup in `zinnia` as part of https://github.com/filecoin-station/zinnia/issues/245

Reply:
I wasn't suggesting always needing to pass `CACHE_ROOT`, I thought if it's not passed then lassie shall pick its own temp dir, if it is passed (as in Station Core) it shall use that, just to keep all of the files together.

The primary use case for changing lassie's temp dir to me is not CLI usage but inside Station.

> Note: if we tell Lassie to use a specific temp dir that's not automatically cleaned by the operating system, we will need to clean any leftover files ourselves.

That's a great point! What does Lassie use by default rn? If it uses an OS cleaned up dir, I'd suggest to leave it at that and add a comment summarizing our discussion here

Reply:
> That's a great point! What does Lassie use by default rn? If it uses an OS cleaned up dir, I'd suggest to leave it at that and add a comment summarizing our discussion here

Quoting from Lassie comments:
https://github.com/filecoin-project/lassie/blob/afc2ee5a4bc6f5e22ef2cc69396cc9b25f57b854/pkg/lassie/lassie.go#L199-L201
```go
// WithTempDir allows you to specify a custom temp directory for bitswap
// retrievals, used for a temporary block store for the preloader. The default
// is the system temp directory.
```

I think that should be good enough for now, even if we may end up leaving some temporary files behind when `zinnia` exists unexpectedly.